src/common/mpu6050.rs

```diff
@@ -0,0 +1,348 @@
+#![allow(dead_code)]
+use crate::common::i2c::I2cHandleType;
+use crate::common::math_utils::Vector3;
+use crate::common::movement_sensor::{MovementSensor, MovementSensorSupportedMethods};
+
+use super::board::{Board, BoardType};
+use super::config::Kind::BoolValue;
+use super::config::{Component, ConfigType};
+use super::i2c::I2CHandle;
+use super::movement_sensor::MovementSensorType;
+use super::registry::ComponentRegistry;
+use super::status::Status;
+
+use byteorder::{BigEndian, ReadBytesExt};
+use std::collections::BTreeMap;
+use std::io::Cursor;
+use std::sync::mpsc::{self, Sender, TryRecvError};
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::Duration;
+
+// This module represents an implementation of the MPU-6050 gyroscope/accelerometer
+// as a Movement Sensor component
+
+pub(crate) fn register_models(registry: &mut ComponentRegistry) {
+    if registry
+        .register_movement_sensor("gyro-mpu6050", &MPU6050::from_config)
+        .is_err()
+    {
+        log::error!("gyro-mpu6050 type is already registered");
+    }
+}
+
+// This is a struct meant to hold the most recently polled readings value
+// and the most recent success or error state in acquiring readings. This
+// is internal to this driver and should be accessed via Mutex.
+#[derive(Clone, Debug)]
+struct MPU6050State {
+    angular_velocity: Vector3,
+    linear_acceleration: Vector3,
+    // temperature: f64,
+    error: Option<Arc<anyhow::Error>>,
+}
+
+impl MPU6050State {
+    fn new() -> Self {
+        MPU6050State {
+            angular_velocity: Vector3::new(),
+            linear_acceleration: Vector3::new(),
+            // temperature: 0.0,
+            error: None,
+        }
+    }
+
+    fn get_angular_velocity(&self) -> Vector3 {
+        self.angular_velocity
+    }
+
+    fn get_linear_acceleration(&self) -> Vector3 {
+        self.linear_acceleration
+    }
+
+    fn set_error(&mut self, err: Option<Arc<anyhow::Error>>) {
+        self.error = err;
+    }
+
+    fn set_linear_acceleration_from_reading(&mut self, reading: &[u8; 14]) {
+        let mut slice_copy = vec![0; 6];
+        slice_copy.clone_from_slice(&(reading[0..6]));
```

Comment:
avoid making a copy here you should be able to convert from the buffer directly (use from_be_bytes/from_le_bytes)